### DISCUSSION TITLE
Any CQRS users, question about disposing DataConnection

### DISCUSSION THREAD

**Discussion by sgashua** [linq2db/linq2db#3416] [Q&A] [answered]
I just started CQRS. I had been doing unit of work & repository pattern and always use dispose in business logic. Now I'm not sure where to put dispose in new CQRS.

I inject dataconnection
```
services.AddLinqToDbContext<DatabaseContext>((provider, options) =>
{
    options.UseSqlServer(configuration.GetConnectionString("DefaultConnection"));
});
```

In GetCopmanyQueryhandler.cs
```
public class GetCompanyQueryHandler : IRequestHandler<GetCompanyQuery, Company>
{
    private readonly DatabaseContext _context;
    public GetCompanyQueryHandler(DatabaseContext context)
    {
        _context = context;
    }

    public async Task<Company> Handle(GetCompanyQuery command, CancellationToken token)
    {
        return _context.Companies.FirstOrDefault(x => x.CompanyID == command.CompanyID);
    }
}
```

Do I still need to put IDisposable / dispose in CQRS? Or not necessary and it will dispose automatically? Anyone advises? 

Thanks.

**Comment by sdanyliv** [comment, marked as answer]
`DatabaseContext` registered as scoped, so it will be disposed at end of the request automatically.

**Reply by sgashua** [reply]
Ok, thanks.
